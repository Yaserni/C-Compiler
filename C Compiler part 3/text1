
function void main()
{
   
    string x[12];
    var char a;
    x[2]=x[3];
    
}

