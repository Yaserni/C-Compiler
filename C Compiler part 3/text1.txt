function int f(int x)
{
    var int a;
    var char aa;
	string s[12];

return aa;
}
function void main()
{
    var int a;
	string s[12];
	if(a>3)
	{
	var int x=a;
	x=a+x-a+x;
			
	}
}
