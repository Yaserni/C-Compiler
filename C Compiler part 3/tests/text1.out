(CODE
    (
        (FUNCTION foo
            (ARGS
                ( INT
                    ( x
                        ( y z)
                        
                    )
                    ( REAL f)
                    
                )
            ) (TYPE VOID)
            (BODY
                (

                    (IF-ELSE
                        (> x y)
                        
                        (BLOCK

                            (= x
                                (+ x f)
                                
                            )
                        )
                    )
                )
            )
        (
            (FUNCTION goo (ARGS NONE) (TYPE CHAR)
                (BODY

                    (VAR INT
                        (= c 0)
                        
                    )
                )
                )
                (RETURN 'a')
                
            (FUNCTION MAIN (ARGS NONE) (TYPE VOID)
                (BODY )
                
            )
        )
    )
)
Accepted
